monai/metrics/seg_metric_utils.py

```diff
@@ -0,0 +1,220 @@
+        input_data = torch.argmax(input_data, dim=1, keepdim=True)
+        input_data = one_hot(input_data, num_classes=n_classes)
+    return input_data
+
+
+def preprocess_input(
+    y_pred: torch.Tensor,
+    y: torch.Tensor,
+    to_onehot_y: bool = False,
+    activation: Optional[Union[str, Callable]] = None,
+    bin_mode: Optional[str] = None,
+    bin_threshold: Union[float, Sequence[float]] = 0.5,
+    include_background: bool = True,
+):
+    """
+    Args:
+        y_pred: predictions. As for classification tasks,
+            `y_pred` should has the shape [BN] where N is larger than 1. As for segmentation tasks,
+            the shape should be [BNHW] or [BNHWD].
+        y: ground truth, the first dim is batch.
+        to_onehot_y: whether to convert `y` into the one-hot format. If not to convert the format, ``y``
+        should has the same shape as ``y_pred``. Defaults to False.
+        activation: [``None``, ``"sigmoid"``, ``"softmax"``]
+            Activation method, if specified, an activation function will be employed for `y_pred` before
+            computation. Defaults to None. The parameter can also be a callable function, for example:
+            ``activation = lambda x: torch.log_softmax(x)``.
+        bin_mode: [``None``, ``"threshold"``, ``"mutually_exclusive"``]
+            Binarization method, if specified, a binarization manipulation will be employed for `y_pred` before
+            computation. Defaults to None.
+
+            - ``"threshold"``, a single threshold or a sequence of thresholds should be set.
+            - ``"mutually_exclusive"``, `y_pred` will be converted by a combination of `argmax` and `to_onehot`.
+        bin_threshold: the threshold for binarization, can be a single value or a sequence of
+            values that each one of the value represents a threshold for a class.
+        include_background: whether to include computation on the first channel of the predicted output.
+            Defaults to True.
+
+    Raises:
+        AssertionError: When ``y_pred`` only has one dimebsion.
+        AssertionError: When ``y`` and ``y_pred`` have different shapes before output.
+
+    """
+
+    # convert y into one-hot version if needed
+    assert y_pred.ndimension() > 1, "y_pred should at least has two dimensions batch and n_classes."
+    num_classes = y_pred.shape[1]
+    if to_onehot_y:
+        y = do_onehot(y, num_classes)
+    # do activation for y_pred if needed
+    if activation is not None:
+        y_pred = do_activation(y_pred, activation=activation)
+    # binarization for y_pred if needed
+    if bin_mode is not None:
+        y_pred = do_binarization(y_pred, bin_mode=bin_mode, bin_threshold=bin_threshold)
+    # remove background if needed
+    if not include_background:
+        y = y[:, 1:] if y.shape[1] > 1 else y
+        y_pred = y_pred[:, 1:] if y_pred.shape[1] > 1 else y_pred
+    assert y.shape == y_pred.shape, "data shapes of y_pred and y do not match."
```

Comment:
This assert makes more sense, being used to check the correctness of an internal condition of your algorithm.